code/datums/components/clothing_dirt.dm

```diff
@@ -0,0 +1,90 @@
+/// This component applies tint to clothing when its exposed to pepperspray, used in /obj/item/clothing/mask/gas.
+
+/datum/component/clothing_dirt
+	/// The ITEM_SLOT_* slot the item is equipped on, if it is.
+	var/equipped_slot
+
+	/// Mob wearing the clothing
+	var/mob/living/carbon/wearer
+
+	/// Amount of dirt stacks on the clothing
+	var/dirtiness = 0
+
+	/// Clothing we're applying tint on
+	var/obj/item/clothing/clothing
+
+/datum/component/clothing_dirt/Initialize()
+	if(!isclothing(parent))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/clothing_dirt/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, PROC_REF(on_equip))
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_drop))
+	RegisterSignal(parent, COMSIG_COMPONENT_CLEAN_ACT, PROC_REF(on_clean))
+	RegisterSignal(parent, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_expose))
+	clothing = parent
+
+/datum/component/clothing_dirt/UnregisterFromParent()
+	clothing.tint -= dirtiness
+	clothing = null
+	if (!isnull(wearer))
+		wearer.update_tint()
+		UnregisterSignal(wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
+		wearer = null
+	else
+		UnregisterSignal(parent, COMSIG_ATOM_EXPOSE_REAGENTS)
+	UnregisterSignal(parent, list(
+		COMSIG_ATOM_EXAMINE,
+		COMSIG_ITEM_EQUIPPED,
+		COMSIG_MOB_UNEQUIPPED_ITEM,
+		COMSIG_COMPONENT_CLEAN_ACT,
+	))
+	return ..()
+
+/datum/component/clothing_dirt/proc/on_equip(datum/source, mob/user, slot)
+	SIGNAL_HANDLER
+	if (!(slot & (ITEM_SLOT_MASK | ITEM_SLOT_HEAD)))
+		return
+	UnregisterSignal(parent, COMSIG_ATOM_EXPOSE_REAGENTS)
+	wearer = user
+	RegisterSignal(wearer, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_expose))
+
+/datum/component/clothing_dirt/proc/on_drop()
+	SIGNAL_HANDLER
+	if(!isnull(wearer))
+		UnregisterSignal(wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
+		wearer = null
+	RegisterSignal(parent, COMSIG_ATOM_EXPOSE_REAGENTS, PROC_REF(on_expose))
+
+/datum/component/clothing_dirt/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	if (dirtiness > 0)
+		examine_list += span_warning("It appears to be covered in some oily substance. Won't see much while wearing it until you wash it off.")
+
+/datum/component/clothing_dirt/proc/on_expose(atom/target, list/reagents, datum/reagents/source, methods)
+	SIGNAL_HANDLER
+	if(QDELETED(wearer) || is_protected() )
+		return
+
+	var/datum/reagent/consumable/condensedcapsaicin/pepper = locate() in reagents
+	if(isnull(pepper))
+		return
```

Comment:
You can (should?) just use `reagents.get_reagent_amount(/datum/reagent/consumable/condensedcapsaicin)`

Which would return 0 if there is none or X if there is, and X can be used in the calculation later `round(x / 5)`

Reply:
that doesn't work, maybe cause the reagents are a list? idk

Reply:
What doesn't work about it? Should be fine, it's how we do it everywhere else

Reply:
it doesn't compile

Reply:
What part of it doesn't compile like show me the compile error

Reply:
tommorow...

Reply:
code\datums\components\clothing_dirt.dm:67:error: reagents.get_reagent_amount: undefined proc

Reply:
now that makes sense